### DISCUSSION TITLE
165

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#712] [Comments]
http://jojoldu.tistory.com/165

**Comment by jojoldu** [comment]
작성자: sejong

혼자서 끙끙대었던 부분인데 정리해주신 내용이 도움많이되었습니다. 감사합니다.

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

도움이 되셨다니 다행입니다^^;
자주 방문 부탁드리겠습니다!
감사합니다!

**Comment by jojoldu** [comment]
작성자: pkgonan

안녕하세요~ 찾고 있던 글인데 도움이 많이 되었습니다.

헌데, Entity Graph를 사용하면 LEFT OUTER JOIN만 작동되나요?
INNER JOIN의 경우에는 방법이 없을가요 허허

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

ㅎㅎ ㅠ 저도 아직 그 부분은 못찾았습니다.
@Query의 join fetch를 추천드립니다 ㅠ
querydsl을 쓸수있다면 더 추천드립니다 ㅠ

**Comment by alsdns016** [comment]
안녕하세요. 위에 소스 중 궁금한게 있어 문의드립니다.

    @Transactional(readOnly = true)
    public List<String> findAllSubjectNames(){
        return extractSubjectNames(academyRepository.findAll());
    }

여기부분을 보면 findAll()한 데이터들을 extractSubjectNames함수를 실행시켜서 
Subject객체의 name값만 추출하게 되어있습니다.

저는 findAll()한 결과값은 전부 출력하고싶습니다.

그래서 다음과 같이 바꾸었습니다.
    @Transactional(readOnly = true)
    public List<Academy> findAllSubjectNames(){
        return academyRepository.findAll();
    }

하지만에러가나요....

쿼리조회 시 여러개를 가져오니까 당연히 List로 받을 수 있다고 생각했지만 그렇지 않더라구요...

조인한 결과값을 모두 출력하려면 어떤방식으로 가져와야하나요?

원하는 결과값은 
Academy{
academy.fields...
Subejct: [{},{}]
}
이렇게.. 

그냥 findAll한 결과값을 Academy클래스형태로 출력하고싶어요.

**Comment by skyhills13** [comment]
안녕하세요. 
작성하신 글과 같이 intellij 상에서 entity들을 모델링된 페이지로 뽑으려면 어떻게 하면 되나요?

**Comment by parkhj1004** [comment]
안녕하세요~~
매번.. 궁금해서 구글링 하면.. 항상.. 이리로 들어오게 되네요~~
매번 드는 생각이지만.. 완전 정리 짱~~~
오늘도 .. 좋은 정보 얻고 가요~^^

**Comment by eari100** [comment]
제가 찾던 내용이네요 감사합니다

**Comment by roeniss** [comment]
`EntityGraph`가 outer join을 유발한다면 가급적 `join fetch`를 쓰는 편이 나을 것 같은데, `EntityGraph`를 쓰는 편이 나을 만한 상황이 있을까요?

**Comment by hackyeah039** [comment]
안녕하세요 JPA  공부중인 허접입니다!
처음 Academy 엔티티를 볼때 궁금한게 

@JoinColumn(name="academy_id")
private List<Subject> subjects = new ArrayList<>();

이 필드를 작성할때 조인칼럼은  mapped by ="academy"  이런식으로 써야하지 않나요?

**Comment by HyundeokPark** [comment]
안녕하세요. JPA N+1 문제 공부를 하면서 글을 보게 되었습니다.

그런데 JoinFetch와 EntityGraph로 실행시 JoinFetch는 카타시안 곱이 나오는데, EntityGraph는 나오지 않아서요 

혹시 왜 그런지 알수있을까요?? 여러군데 검색해봤지만 명확한 이유를 모르겠네요...
